lib/galaxy/jobs/__init__.py

```diff
@@ -1770,10 +1777,14 @@ def reclaim_ownership( self ):
     def user_system_pwent( self ):
         if self.__user_system_pwent is None:
             job = self.get_job()
-            try:
-                self.__user_system_pwent = pwd.getpwnam( job.user.email.split('@')[0] )
-            except:
-                pass
+
+            # try to get user from email address/username
+            for cand in [job.user.email.split('@')[0], job.user.username]:
```

Comment:
For adding options to galaxy.ini you'd write your option in the galaxy.ini.sample (with a bit of documentation) and then you reference it in the config object [here](https://github.com/galaxyproject/galaxy/blob/dev/lib/galaxy/config.py#L349). This will then be available as app.config.your_new_option (in this class it's `self.app.config.your_new_option`)